src/Microsoft.Azure.SignalR.AspNet/Infrastructure/SignalRMessageParser.cs

```diff
@@ -0,0 +1,216 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.AspNet.SignalR;
+using Microsoft.AspNet.SignalR.Infrastructure;
+using Microsoft.AspNet.SignalR.Json;
+using Microsoft.AspNet.SignalR.Messaging;
+using Microsoft.AspNet.SignalR.Transports;
+using Microsoft.Azure.SignalR.Protocol;
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.SignalR.AspNet
+{
+    internal class SignalRMessageParser : IMessageParser
+    {
+        private const char DotChar = '.';
+
+        private readonly HashSet<string> _hubNameWithDots = new HashSet<string>();
+
+        private readonly JsonSerializer _serializer;
+        private readonly IMemoryPool _pool;
+        private readonly IServiceProtocol _serviceProtocol;
+
+        public SignalRMessageParser(IReadOnlyList<string> hubs, IDependencyResolver resolver)
+        {
+            _serializer = resolver.Resolve<JsonSerializer>() ?? throw new ArgumentNullException(nameof(JsonSerializer));
+            _serviceProtocol = resolver.Resolve<IServiceProtocol>() ?? throw new ArgumentNullException(nameof(IServiceProtocol));
+            _pool = resolver.Resolve<IMemoryPool>() ?? throw new ArgumentNullException(nameof(IMemoryPool));
+
+            foreach(var hub in hubs)
```

Comment:
Add a comment about what these hub names look like. Are these the full type name in the default case?

Reply:
They are fetched from `IHubManager.GetHubs()`'s `Name` property, and by default it is the type `Name` instead of `FullName`

Reply:
Can you add that comment?

Reply:
sure